property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
+      case 'string':
+        if (value === 'uuid' || value === 'guid'){
+          def.defaultFn = value;
+        } else {
+          def.default = value;
+        }
+        break;
+      case 'number':
+        def.default = Number(value);
+        break;
+      case 'boolean':
+        if (['true', '1', 't'].indexOf(value) !== -1 ){
+          def.default = true;
+        } else {
+          def.default = false;
+        }
+        break;
+      case 'object':
+        def.default = JSON.parse(value);
+        break;
+      case 'array':
+        def.default = value.replace(/[\s,]+/g, ',').split(',');
+        break;
+      case 'date':
+        console.warn('Warning: property default value was converted' +
+          ' from string using ISO formatting');
+        if (value.toLowerCase() === 'now'){
+          def.defaultFn = 'now';
+        } else {
+          def.default = new Date(value);
```

Comment:
Note that `new Date()` supports a wide range of different formats, not only ISO formatting. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse

It would be nice to support numeric timestamps for default values too. E.g. if the value matches `/^[0-9]+$/`, then we should convert it to a number before passing it to `Date` constructor.

Notice that `new Date('0') !== new Date(0)`.